src/core/backend_context.cc

```diff
@@ -478,7 +513,9 @@ BackendResponder::FlushPendingPinned(
   if (pinned_memory_type != TRITONSERVER_MEMORY_CPU_PINNED) {
```

Comment:
In this case here we didn't allocate a pinned buffer so we are doing CPU->GPU or GPU->CPU direct, and so we don't want any of the async copy changes below.